src/sys/socket/addr.rs

```diff
@@ -23,15 +23,176 @@ use ::sys::socket::addr::sys_control::SysControlAddr;
 #[repr(i32)]
```

Comment:
Can you also delete the comments above and the `TODO` (which I think is actually resolved also)? It'd be good to clean those up since you're already in here.